pyccel/codegen/printing/ccode.py

```diff
@@ -1202,13 +1206,37 @@ def _print_Constant(self, expr):
     def _print_Return(self, expr):
         code = ''
         args = [VariableAddress(a) if self.stored_in_c_pointer(a) else a for a in expr.expr]
+
+        if len(args) > 1:
+            if expr.stmt:
+                return self._print(expr.stmt)+'\n'+'return 0;'
+            return 'return 0;'
+
         if expr.stmt:
-            code += self._print(expr.stmt)+'\n'
-        if len(args) == 1:
-            code +='return {0};'.format(self._print(args[0]))
-        elif len(args) > 1:
-            code += 'return 0;'
-        return code
+            # get Assign nodes form CodeBlock object expr.stmt
+            assigns = expr.stmt.get_attribute_nodes(Assign)
+
+            # check the Assign objects list in case of
+            # the user assigns a variable to an object contains IndexedElement object
+            if not assigns:
+                return 'return {0};'.format(self._print(args[0]))
+            last_assign = assigns[-1]
+            deallocations = expr.stmt.get_attribute_nodes(Deallocate)
+
+            # print deallocation nodes
+            if deallocations:
+                code += '\n' + self._print(last_assign)
+                for a in deallocations:
+                    code += '\n' + self._print(a)
```

Comment:
Can `Return.stmt` contain anything other than deallocates and an extra assign? Currently there is nothing stopping us including anything else so I'm not sure we can definitely throw away anything else. Could we not simply do:
```
if last_assign.lhs.is_temp:
    code = '\n'.join(l for l in return.stmt if l is not last_assign)
    code += '\nreturn {}'.format(last_assign.rhs)
else:
    code = '\n'.join(l for l in return.stmt)
    code += '\nreturn {}'.format(return.val)
```
?
(Sorry my code is not 100% correct or neat, I'm on my phone. Hopefully it's enough to get the idea)

Reply:
sorry, i forgot to use `is_temp` attribute, it is really useful in this case